relayer/src/link/relay_path.rs

```diff
@@ -1438,62 +1598,90 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
             (true_res, false_res)
         }
 
-        let connection_delay = self.channel.connection_delay;
+        let src_chain_time = self.src_time_latest()?;
+        let src_max_block_time = self.src_max_block_time()?;
+        let src_latest_height = self.src_latest_height()?;
         let (elapsed_src_ods, unelapsed_src_ods) =
             partition(self.src_operational_data.take(), |op| {
-                op.scheduled_time.elapsed() > connection_delay
+                op.conn_time_delay_elapsed(src_chain_time).is_ok()
+                    && op
+                        .conn_block_delay_elapsed(src_max_block_time, src_latest_height)
+                        .is_ok()
             });
 
-        self.src_operational_data.replace(unelapsed_src_ods);
-
+        let dst_chain_time = self.dst_time_latest()?;
+        let dst_max_block_time = self.dst_max_block_time()?;
+        let dst_latest_height = self.dst_latest_height()?;
         let (elapsed_dst_ods, unelapsed_dst_ods) =
             partition(self.dst_operational_data.take(), |op| {
-                op.scheduled_time.elapsed() > connection_delay
+                op.conn_time_delay_elapsed(dst_chain_time).is_ok()
+                    && op
+                        .conn_block_delay_elapsed(dst_max_block_time, dst_latest_height)
+                        .is_ok()
             });
 
+        self.src_operational_data.replace(unelapsed_src_ods);
         self.dst_operational_data.replace(unelapsed_dst_ods);
-
-        (elapsed_src_ods, elapsed_dst_ods)
+        Ok((elapsed_src_ods, elapsed_dst_ods))
     }
 
     /// Fetches an operational data that has fulfilled its predefined delay period. May _block_
     /// waiting for the delay period to pass.
     /// Returns `None` if there is no operational data scheduled.
```

Comment:
This doc comment needs to be updated to reflect this function's new return type.

Reply:
https://github.com/informalsystems/ibc-rs/pull/1917/commits/d1b712b9dbcc847bbd10133bef0064653a5f3617